File: kinetiaServer/src/main/java/com/proyectoi/kinetia/domain/Message.java
package com.proyectoi.kinetia.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.proyectoi.kinetia.models.MessageModel;

import java.time.Instant;


public class Message {

    @JsonProperty("id")
    private Long id;
    @JsonProperty("sender")
    private Long sender;
    @JsonProperty("recipient")
    private Long recipient;
    @JsonProperty("content")
    private String content;
    @JsonProperty("sentAt")
    private Instant sentAt;
    @JsonProperty("isRead")
    private Boolean isRead;

    public Message() {}

    public Message(MessageModel message) {
        this.id = message.getId();
        this.sender = message.getSender().getId();
        this.recipient = message.getReceiver().getId();
        this.content = message.getContent();
        this.sentAt = message.getSentAt();
        this.isRead = message.getIsRead();
    }

    public Long getSender() {
        return sender;
    }

    public Long getRecipient() {
        return recipient;
    }

    public String getContent() {
        return content;
    }

    public Instant getSentAt() {
        return sentAt;
    }

    public Boolean getIsRead() {
        return isRead;
    }
}

File: kinetiaApp/app/src/main/java/com/dam2/proyectocliente/models/Message.kt
package com.dam2.proyectocliente.models

import com.dam2.proyectocliente.utils.InstantSerializer
import kotlinx.serialization.Serializable
import java.time.Instant

@Serializable
data class Message(
    var id: Long? = null,
    val sender: Long,
    val recipient: Long,
    val content: String,
    @Serializable(with = InstantSerializer::class)
    val sentAt: Instant = Instant.now(),
    var isRead: Boolean = false

)


